packages/layers/src/xr-layer/xr-layer.js

```diff
@@ -245,13 +263,22 @@ const XRLayer = class extends Layer {
         domain,
         dtype
       });
-      model
-        .setUniforms({
+      //HACK: null textures will throw errors, so we just set them all to the first texture FOR THE VERY SHORT TERM!
+      for (const key in textures) {
+        if (!textures.channel0) throw new Error('Bad texture state!');
+        if (!textures[key]) textures[key] = textures.channel0;
+      }
+      model.setUniforms(
+        {
           ...uniforms,
-          contrastLimits: paddedContrastLimits,
-          ...textures
-        })
-        .draw();
+          contrastLimits: paddedContrastLimits
+        },
+        { disableWarnings: false }
+      );
+      model.setBindings(textures);
+      // is `opts.uniforms` still the same as what was passed in?
+      // seems to get the right result, but I'm unclear on the implications of this
```

Comment:
Docs aren't amazingly clear but it seems to be default `uniforms` from `opts`? https://deck.gl/docs/api-reference/core/layer#draw. And then state model is used for drawing

Reply:
So I think this comment can be removed (I only see the two listed here: https://deck.gl/docs/developer-guide/custom-layers/writing-shaders#uniforms)

Reply:
```suggestion
```